thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/FilterList.kt

```diff
@@ -1,15 +1,24 @@
 package io.github.paulgriffith.kindling.thread
 
 import com.jidesoft.swing.CheckBoxList
+import com.jidesoft.swing.ListSearchable
 import io.github.paulgriffith.kindling.utils.NoSelectionModel
 import io.github.paulgriffith.kindling.utils.listCellRenderer
 import java.text.DecimalFormat
 import javax.swing.AbstractListModel
 import javax.swing.ListModel
 
+typealias FilterComparator = Comparator<Map.Entry<String?, Int>>
+
 class FilterModel(val rawData: Map<String?, Int>) : AbstractListModel<Any>() {
-    private val comparator: Comparator<Map.Entry<String?, Int>> = compareByDescending(nullsFirst()) { it.value }
-    private val values = rawData.entries.sortedWith(comparator).map { it.key }
+    var comparator: FilterComparator = byCountDesc
+        set(value) {
+            values = rawData.entries.sortedWith(comparator).map { it.key }
+            fireContentsChanged(this, 0, size)
+            field = value
+        }
```

Comment:
Now we don't need to construct a new model - just change the comparator and everything else figures itself out